base_layer/core/src/transactions/aggregated_body.rs

```diff
@@ -193,6 +195,86 @@ impl AggregateBody {
         fee
     }
 
+    /// This function will check spent kernel rules like tx lock height etc
+    pub fn check_kernel_rules(&self, height: u64) -> Result<(), TransactionError> {
+        for kernel in self.kernels() {
+            if kernel.lock_height > height {
+                debug!(target: LOG_TARGET, "Kernel lock height was not reached: {}", kernel);
+                return Err(TransactionError::InvalidKernel);
+            }
+        }
+        Ok(())
+    }
+
+    /// Run through the outputs of the block and check that
+    /// 1. There is exactly ONE coinbase output
+    /// 1. The output's maturity is correctly set
+    /// 1. The amount is correct.
+    pub fn check_coinbase_output(
+        &self,
+        reward: MicroTari,
+        consensus_constants: &ConsensusConstants,
```

Comment:
Rather take lock_height as an argument here, then you can drop the namespace import of ConsensusConstants AND it would make testing easier since you wouldn't have to create a whole CC object to test this fn.

Reply:
done